src/desktop/Workspace.cpp

```diff
@@ -648,6 +648,9 @@ void CWorkspace::rename(const std::string& name) {
     Debug::log(LOG, "CWorkspace::rename: Renaming workspace {} to '{}'", m_iID, name);
     m_szName = name;
 
+    const auto WORKSPACERULE = g_pConfigManager->getWorkspaceRuleFor(m_pSelf.lock());
+    m_bPersistent            = WORKSPACERULE.isPersistent;
```

Comment:
wouldnt it be better to call recheck persistent?

Reply:
Sure, I see that it's a convention in the project. And how do you feel about also calling updateWindowData from the renaming handler?